go-controller/pkg/ovn/base_network_controller_secondary.go

```diff
@@ -337,6 +338,46 @@ func (bsnc *BaseSecondaryNetworkController) addLogicalPortToNetworkForNAD(pod *k
 		ops = append(ops, addOps...)
 	}
 
+	// Pods on user defined secondary networks should be able to communicate
+	// to the outside.  Add SNATs for them.
+	if util.IsNetworkSegmentationSupportEnabled() && bsnc.IsPrimaryNetwork() {
+		if config.Gateway.DisableSNATMultipleGWs {
```

Comment:
This should be removed `if config.Gateway.DisableSNATMultipleGWs {` I think. We dont want to respect this for role primary secondary networks.

Reply:
@tssurya right?

Reply:
commit removed